corehq/apps/users/models.py

```diff
@@ -2282,17 +2287,18 @@ def get_usercase_id(self):
         case = self.get_usercase()
         return case.case_id if case else None
 
-    def update_device_id_last_used(self, device_id, when=None, commcare_version=None, device_app_meta=None):
+    def update_device_id_last_used(self, device_id, when=None, commcare_version=None, device_app_meta=None,
+                                   fcm_token=None, fcm_token_timestamp=None):
         """
         Sets the last_used date for the device to be the current time
         Does NOT save the user object.
 
         :returns: True if user was updated and needs to be saved
         """
         when = when or datetime.utcnow()
-
         device = self.get_device(device_id)
         if device:
+            device.update_fcm_token(fcm_token, fcm_token_timestamp)
```

Comment:
> also, you might need to consider setting `do_update=True` here? or the device record will get updated anyway?

Thank you for catching this!

Reply:
@mkangia 
Resolved in https://github.com/dimagi/commcare-hq/pull/33047/commits/7451c71fe1b0afd94b79f44ef3a162893fa4312d
Note: `do_update`  has a different setup of conditions to update device meta data.
In order to avoid not changing those conditions , have used `save_user` flag which becomes true if we update fcm token.